packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -1043,7 +1046,7 @@ class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _Tap
 
         // This can occur when the recognizer is accepted before a [PointerMoveEvent] has been
         // received that moves the pointer a sufficient global distance to be considered a drag.
-        if (_start != null && _sentTapDown) {
```

Comment:
Just a question for my learning, why do we remove `_sentTapDown` here?

Reply:
One of the drag callbacks may leak through if `_sentTapDown` is in this conditional. Same reason why `TextField-onTap` leaked (`_sentTapDown` is switched on when the recognizer wins the arena or the deadline of `kPressTimeout` is exceeded, we only want to let other callbacks pass through when we win the arena).